drivers: modem: hl7800: Add runtime control of log level

rerickson1 opened the pull request:
Default mode must be debug or it can't ever be enabled.
Allow debug log messages to be printed when using
mdm_hl7800_send_at_cmd API.
Add logging to active bands and network coverage commands.

rlubos commented on drivers/modem/hl7800.c:
```diff
@@ -1070,6 +1072,32 @@ void mdm_hl7800_generate_status_events(void)
+	return -1;
+}
+
+uint32_t mdm_hl7800_log_filter_set(uint32_t level)
```
Who's the user of this function, responsible for changeing the log level? I see in the commit message that the `shell` is involved - why not use the shell module provided by the logger to change the log level?

rerickson1 replied:
I will fix the commit message. The goal is that any app can change the log level output at runtime even if the shell isn't being used.

rlubos commented on drivers/modem/hl7800.c:
```diff
@@ -1070,6 +1072,32 @@ void mdm_hl7800_generate_status_events(void)
 }
 
+/** @ref /zephyr/samples/subsys/logging/logger/src/main.c */
+static int log_source_id_get(const char *name)
```
Perhaps it'd be better to expose this functionality from the logger module itself instead of copying the function around? Similar is used not only in the mentioned sample, but in `log_cmds.c` as well. @nordic-krch Any opinion on this?

carlescufi replied:
I agree with @rlubos here. @nordic-krch could you also weigh in please?

rerickson1 replied:
@nordic-krch any thoughts? I'd like to get some movement on this. Thanks!

nashif commented:
@rerickson1  please rebase

mbolivar-nordic commented:
@nordic-krch ping

rerickson1 committed:
drivers: modem: hl7800: Add runtime control of log level

Default mode must be debug or it can't ever be enabled.
Allow debug log messages to be printed when using
mdm_hl7800_send_at_cmd API.
Add logging to active bands and network coverage commands.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>

nordic-krch commented on include/zephyr/logging/log_ctrl.h:
```diff
@@ -120,6 +120,15 @@ const char *log_source_name_get(uint32_t domain_id, uint32_t source_id);
+ *
+ * @param name Source name
+ *
+ * @return Source ID.
```
or negative number when source ID is not found.

nordic-krch reviewed (dismissed):
looks ok, one minor comment.

rerickson1 committed:
logging: add log_source_id_get

Add API to get the logging module source ID from the
module name;

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>
